olive/cli/auto_opt.py

```diff
@@ -0,0 +1,255 @@
+                " default to cpu device with CPUExecutionProvider."
+            ),
+        )
+        system_group.add_argument(
+            "--providers",
+            type=str,
+            nargs="*",
+            choices=["CPU", "CUDA", "Tensorrt", "Dml", "VitisAI", "Qnn"],
+            help="List of execution providers to use for optimization",
+        )
+
+        # dataset options
+        dataset_group = sub_parser.add_argument_group(
+            "dataset options, required for some optimization passes like quantization, and evaluation components"
+        )
+        dataset_group.add_argument(
+            "--data_config_path",
+            type=str,
+            help="Path to the data config file. It allows to customize the data config(json/yaml) for the model.",
+        )
+
+        auto_opt_config_group = sub_parser.add_argument_group("auto optimizer options")
+        auto_opt_config_group.add_argument(
+            "--precisions",
+            type=str,
+            nargs="*",
+            choices=["fp16", "fp32", "int4", "int8"],
+            help=(
+                "The output precision of the optimized model. If not specified, "
+                "the default precision is fp32 for cpu and fp16 for gpu"
+            ),
+        )
+        auto_opt_config_group.add_argument(
+            "--excluded_passes",
+            type=str,
+            nargs="*",
+            help=(
+                "List of passes to disable for optimization, if not specified, "
+                "auto-opt will disable ModelBuilder/OrtPerfTuning by default."
+            ),
+        )
+        auto_opt_config_group.add_argument(
+            "--use_model_builder",
+            action="store_true",
+            help=(
+                "Whether to use model builder pass for optimization, enable only "
+                "when the model is supported by model builder"
+            ),
+        )
+
+        search_strategy_group = sub_parser.add_argument_group("search strategy options")
+        search_strategy_group.add_argument(
+            "--num_samples", type=int, default=5, help="Number of samples for search algorithm"
+        )
+        search_strategy_group.add_argument("--seed", type=int, default=0, help="Random seed for search algorithm")
+        search_strategy_group.add_argument(
+            "--search_order",
+            type=str,
+            default="joint",
+            choices=["joint", "pass-by-pass"],
+            help="Execution order for search strategy",
+        )
+        search_strategy_group.add_argument(
+            "--search_algorithm",
+            type=str,
+            default="tpe",
+            choices=["exhaustive", "tpe", "random"],
+            help="Search algorithm for search strategy",
+        )
+
+        # output options
+        output_group = sub_parser.add_argument_group("output options")
+        output_group.add_argument(
```

Comment:
Not related to this PR but we could gather this to base.py as well.

Reply:
Good call. But I saw there are different arguments attributes. Some of them are required, some of them are not. And they have different default values. 

Can we update it in follow-up PR?